controllers/cert-manager/v1alpha1_add_label_controller.go

```diff
@@ -0,0 +1,135 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	certmanagerv1 "github.com/ibm/ibm-cert-manager-operator/apis/cert-manager/v1"
+	certmanagerv1alpha1 "github.com/ibm/ibm-cert-manager-operator/apis/certmanager/v1alpha1"
+	res "github.com/ibm/ibm-cert-manager-operator/controllers/resources"
+)
+
+// V1Alpha1AddLabelReconciler reconciles a Certificate object
+type V1Alpha1AddLabelReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=cert-manager.io,resources=certificates,verbs=get;list;watch;create;update;patch
+//+kubebuilder:rbac:groups=cert-manager.io,resources=certificates/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Certificate object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *V1Alpha1AddLabelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logd = log.FromContext(ctx)
+
+	reqLogger := logd.WithValues("req.Namespace", req.Namespace, "req.Name", req.Name)
+	reqLogger.Info("Reconciling CertificateRefresh")
+
+	v1cert := &certmanagerv1.Certificate{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, v1cert)
+	if err == nil {
+		// the corresponding v1 cert exists
+		return ctrl.Result{}, nil
+	} else if !errors.IsNotFound(err) {
+		logd.Error(err, "Error getting v1 Certificate")
+		return ctrl.Result{}, err
+	}
+
+	cert := &certmanagerv1alpha1.Certificate{}
+	err = r.Client.Get(context.TODO(), req.NamespacedName, cert)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logd.Error(err, "Error getting v1alpha1 Certificate")
+		return ctrl.Result{}, err
+	}
+
+	// Get secret corresponding to the certificate
+	secretInstance, err := r.getSecret(cert)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		logd.Error(err, "Error getting Secret")
+		return ctrl.Result{}, err
+	}
+
+	oldLabelsMap := secretInstance.GetLabels()
+	if oldLabelsMap == nil {
+		oldLabelsMap = make(map[string]string)
+	}
+
+	if _, ok := oldLabelsMap[res.SecretWatchLabel]; !ok {
+		oldLabelsMap[res.SecretWatchLabel] = ""
+		secretInstance.SetLabels(oldLabelsMap)
+	}
+
+	r.updateSecret(secretInstance)
+	return ctrl.Result{}, nil
```

Comment:
Need to do something with the error
```suggestion
	return ctrl.Result{}, r.updateSecret(secretInstance)
```